transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/networkinterface/resolution/NetworkInterfaceResolutionIT.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2007-2015, Kaazing Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.gateway.transport.wsn.networkinterface.resolution;
+import static org.kaazing.test.util.ITUtil.createRuleChain;
+
+import java.net.URI;
+
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.kaazing.gateway.server.test.GatewayRule;
+import org.kaazing.gateway.server.test.config.GatewayConfiguration;
+import org.kaazing.gateway.server.test.config.builder.GatewayConfigurationBuilder;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+
+public class NetworkInterfaceResolutionIT {
+    
+    private GatewayRule gateway = new GatewayRule() {
+        {
+            // @formatter:off
+            GatewayConfiguration configuration =
+                    new GatewayConfigurationBuilder()
+                        .cluster()
+                        	//TODO: Add support for strings in cluster accepts
+                            .accept("tcp://[@Software Loopback Interface 1]:2345")
+                            .connect(URI.create("tcp://localhost:5432"))
+                            .name("clusterName")
+                        .done()
+                        .service()
+                            .accept("tcp://localhost:8003")
+                                .acceptOption("tcp.bind", "[@Software Loopback Interface 1]:7082")
+                            .connect(URI.create("tcp://localhost:8004"))
+                            .type("proxy")
+                        .done()
+//                        .service()
+//                            .accept("http://localhost:8110")
+//                                .acceptOption("http.transport", "tcp//[@Software Loopback Interface 1]:7081")
+//                            .connect(URI.create("http://localhost:8080"))
+//                            .type("proxy")
+//                        .done()
+                        .service()
+                            .accept("tcp://[@Software Loopback Interface 1]:8001")
```

Comment:
@jitsni Hello, Jitu, I think the above comment has a major impact on the existing pull request. As I undertand, this would impact:
>
Change ResourceAddressFactory, ResourceAddressFactorySpi methods that take URI parameter to String

- impact on the implementation

>
Similarly in GatewayConfigBuilder accept, connect methods

- impact on the existing testing framework together with all the ITests which make use of the gateway configuration

>
TcpResourceAddressFactorySpi need to be changed to take care of String location with network interface addresses (add alternate addresses if an interface maps to multiple IP addresses)

- impact on the implementation

Can we schedule a meeting to discuss this? As I understand, this would cancel the current implementation approach which performs resolution in the GatewayContextResolver. Thank you.